clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesApiConverter.groovy

```diff
@@ -809,8 +809,17 @@ class KubernetesApiConverter {
   static ReplicaSet toReplicaSet(ReplicaSetBuilder serverGroupBuilder,
                                  DeployKubernetesAtomicOperationDescription description,
                                  String replicaSetName) {
-
-    def targetSize = description.targetSize ?: description.capacity?.desired
+    /*
+    The elvis operator before was not evaluating truthiness correctly for a value of 0.
+    A User could and may want to deploy an update ReplicaSet of 0 and Scale it later.  Or perhaps it's
+    bundled in a Spinnaker Pipeline with multiple Server Groups and one of the Server Groups is scaled to 0.
+    If the User is implementing 'useSourceCapacity' we would want that 0 value respected and not overwritten with a
+    value of 1, which will happen if the targetSize is represented by a null value at this juncture in the code.
+     */
+    Integer targetSize = description.targetSize.toInteger()
+    if (targetSize < 0) {
```

Comment:
Since `targetSize` can never be `null` or `< 0`, I don't think `description.capacity` will ever be considered.

Reply:
Yeah, I'll need to reason about on this... this will need to change.

Reply:
I'm thinking the right way to approach this would be the following:

Defer to the desired source capacity, so if the targetSize is less than it - the desired source capacity wins.  This still covers the use case of this PR as you wouldn't set a desired source capacity of > 0 if you wanted a targetSize of 0 or you wanted the resource at 0.  Essentially, desired source capacity always trumps targetSize.  Thoughts?